txnsync/bitmask.go

```diff
@@ -130,7 +130,11 @@ func (b *bitmask) iterate(entries int, maxIndex int, callback func(int, int) err
 					if index >= maxIndex {
 						return errDataMissing
 					}
-					if err := callback(8*i+j, index); err != nil {
+					ij8 := 8*i + j
+					if ij8 >= entries {
```

Comment:
good catch, maybe we can add a statement outside the for loop that checks that `(len(b)-1)*8` cannot be greater than `entries`, for both case 0 and case 1?

Reply:
it will legitimately be greater for 0 within a byte. It should not be the case for a 1.

Reply:
Putting a statement outside will not save much, and will complicate the logic here.
The savings are limited because anyway within a byte, we need to check to know at which bit we need to stop.